references/cli/addon.go

```diff
@@ -558,21 +564,21 @@ func enableAddon(ctx context.Context, k8sClient client.Client, dc *discovery.Dis
 			}
 			input := NewUserInput()
 			if input.AskBool(unMatchErr.Error(), &UserInputOptions{AssumeYes: false}) {
-				err = pkgaddon.EnableAddon(ctx, name, availableVersion, k8sClient, dc, apply.NewAPIApplicator(k8sClient), config, registry, args, nil)
+				err = pkgaddon.EnableAddon(ctx, addonName, availableVersion, k8sClient, dc, apply.NewAPIApplicator(k8sClient), config, registry, args, nil, pkgaddon.FilterDependencyRegistries(i, registries))
 				return err
 			}
 			// The user does not agree to use the version provided by us
-			return fmt.Errorf("you can try another version by command: \"vela addon enable %s --version <version> \" ", name)
+			return fmt.Errorf("you can try another version by command: \"vela addon enable %s --version <version> \" ", addonName)
 		}
 		if err != nil {
 			return err
 		}
-		if err = waitApplicationRunning(k8sClient, name); err != nil {
+		if err = waitApplicationRunning(k8sClient, addonName); err != nil {
 			return err
 		}
 		return nil
 	}
-	return fmt.Errorf("addon: %s not found in registries", name)
+	return fmt.Errorf("addon: %s not found in registries", addonName)
```

Comment:
We should provide different error message according to the registry.

If specified, report:
```
fmt.Errorf("addon: %s not found in registry %s", addonName, registrName)
```

If not specified, report:
```
fmt.Errorf("addon: %s not found in all candidate registries", addonName)
```
 addon: %s not found in registries